lib/worker/src/db/publications.rs

```diff
@@ -70,17 +70,40 @@ pub async fn is_namespace_owner(pool: &PgPool, ns: &str, owner: Address) -> Resu
     Ok(!res.is_empty())
 }
 
-/// Returns cache config
-pub async fn get_cache_config(pool: &PgPool, ns: &str, rel: &str) -> Result<Option<i64>> {
+/// Returns cache config from a vault
+pub async fn get_cache_config(pool: &PgPool, vault: &Vault) -> Result<Option<i64>> {
     let (duration, ) : (Option<i64>, ) = sqlx::query_as("SELECT duration FROM cache_config JOIN namespaces ON ns_id = namespaces.id WHERE name = $1 AND relation = $2")
-        .bind(ns)
-        .bind(rel)
+        .bind(vault.namespace())
+        .bind(vault.relation())
         .fetch_one(pool)
         .await
         .unwrap_or((None, ));
     Ok(duration)
 }
 
+pub async fn find_cache_config_by_vaults(
```

Comment:
new db method that fetches all configs at once for a list of vaults. the query is a bit complicated because of how the schema of the database is structured